### PR TITLE
Additional trace logging for ESSingleNodeTestCase failures

### PR DESCRIPTION
**Pull request by pmpailis**
Relates to https://github.com/elastic/elasticsearch/issues/110786

### CODE REVIEW

**Comment by pmpailis** [comment]
@elasticmachine update branch

**Review by carlosdelest — APPROVED** [review]
I hope you catch that 🐛 !

File: server/src/main/java/org/elasticsearch/search/SearchService.java [hunk 1]
```diff
@@ -463,6 +464,7 @@ public void beforeIndexShardCreated(ShardRouting routing, Settings indexSettings
     }
 
     protected void putReaderContext(ReaderContext context) {
+        logger.trace("adding reader context [{}]", context.id());
```
Comment (benwtrent) [inline server/src/main/java/org/elasticsearch/search/SearchService.java:467]: I don't think we want to accidentally merge this into main. The test seems to fail enough that it should be reproducible locally or with this PR if the test is unmuted.

Comment (pmpailis) [reply server/src/main/java/org/elasticsearch/search/SearchService.java:467]: Hmm fair enough. Would you be worried of potential impact with this addition? 
Tbh, I haven't been lucky locally even after a huge number of runs. Will try to reproduce within this PR :)

**Comment by pmpailis** [comment]
buildkite test this please

**Comment by pmpailis** [comment]
buildkite test this please

**Comment by pmpailis** [comment]
buildkite test this please

**Comment by pmpailis** [comment]
buildkite test this please

**Comment by pmpailis** [comment]
Closing as a fix for this has been merged through https://github.com/elastic/elasticsearch/pull/111153.

**Pull request closed without merging**